target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/Makefile

```diff
@@ -2,12 +2,12 @@
 # Makefile for the Atheros AR71xx built-in ethernet macs
 #
 
-ag71xx-y	+= ag71xx_main.o
-ag71xx-y	+= ag71xx_gmac.o
-ag71xx-y	+= ag71xx_ethtool.o
-ag71xx-y	+= ag71xx_phy.o
+ag71xx_ds-y	+= ag71xx_main.o
+ag71xx_ds-y	+= ag71xx_gmac.o
+ag71xx_ds-y	+= ag71xx_ethtool.o
+ag71xx_ds-y	+= ag71xx_phy.o
 
-ag71xx-$(CONFIG_AG71XX_DEBUG_FS)	+= ag71xx_debugfs.o
+ag71xx_ds-$(CONFIG_AG71XX_LEGACY_DEBUG_FS)	+= ag71xx_ds_debugfs.o
 
-obj-$(CONFIG_AG71XX)	+= ag71xx_mdio.o
-obj-$(CONFIG_AG71XX)	+= ag71xx.o
+obj-$(CONFIG_AG71XX_LEGACY)	+= ag71xx_ds_mdio.o
```

Comment:
I think we should merge `ag71xx_ds_mdio.ko`  into `ag71xx_ds.ko` I do not see any reson why this should be a seperate module.

Reply:
Is there any good reason to change the code of the legacy driver, or should all the effort be put into implementing the missing parts in the upstream driver?

Reply:
An extra kernel module needs some 100 bytes more space compared to compiling it into one. We always use both modules together.